packages/wallet-sdk/src/sign/scw/utils/createSubAccountSigner.ts

```diff
@@ -52,74 +70,198 @@ export async function createSubAccountSigner({ chainId }: { chainId: number }) {
     factoryData: subAccount.factoryData,
   });
 
-  return {
-    request: async (
-      args: RequestArguments
-    ): Promise<string | Hex | Address[] | number | SubAccount> => {
-      switch (args.method) {
-        case 'wallet_addSubAccount':
-          return subAccount;
-        case 'eth_accounts':
-          return [subAccount.address] as Address[];
-        case 'eth_coinbase':
-          return subAccount.address;
-        case 'net_version':
-          return chainId.toString();
-        case 'eth_chainId':
-          return numberToHex(chainId);
-        case 'eth_sendTransaction': {
-          assertArrayPresence(args.params);
-          return account.sign(args.params[0] as { hash: Hex });
+  const request: EIP1193RequestFn<
+    [...WalletRpcSchema, SendPreparedCallsSchema, PrepareCallsSchema]
+  > = (async (args) => {
+    switch (args.method) {
+      case 'wallet_addSubAccount':
+        return subAccount!;
+      case 'eth_accounts':
+        return [subAccount!.address] as Address[];
+      case 'eth_coinbase':
+        return subAccount!.address;
+      case 'net_version':
+        return chainId.toString();
+      case 'eth_chainId':
+        return numberToHex(chainId);
+      case 'eth_sendTransaction': {
+        assertArrayPresence(args.params);
+        return account.sign(args.params[0] as { hash: Hex });
+      }
+      case 'wallet_sendCalls': {
+        assertArrayPresence(args.params);
+        // Get the client for the chain
+        const chainId = get(args.params[0], 'chainId') as number;
+        assertPresence(chainId, standardErrors.rpc.invalidParams('chainId is required'));
+
+        const walletClient = getClient(chainId);
+        assertPresence(walletClient, standardErrors.rpc.internal('wallet client not found'));
+
+        if (!args.params[0]) {
+          throw standardErrors.rpc.invalidParams('params are required');
+        }
+
+        if (!('calls' in args.params[0])) {
```

Comment:
seems like the current code pattern is to use `get()` for deep access. Let's keep for consistency
```suggestion
        if (!get(args.params[0], 'calls')) {
```